### PR TITLE
Use Giomm dbus instead of dbus-c++

### PR DESCRIPTION
**Pull request by DunderRoffe**
This pull-request attempts to remove dbus-c++ and replace it with giomm dbus bindings instead.
The stubs for the new bindings were generated with a giomm dbus binding generator. However, it was not complete so that some changes had to be made later. In order to make transition, if the generator is patched, to generating at build time, I have chosen to change as little as possible of the generated code. Due to this fact there might be some redundant code and I ask you to please have that in mind when reviewing the pr.

Signed-off-by: Viktor Sjölind <viktor.sjolind@pelagicore.com>

### CODE REVIEW

File: agent/src/main.cpp [hunk 1]
```diff
@@ -237,12 +203,8 @@ int main(int argc, char **argv)
         Config config(std::move(loader), std::move(defaults), stringOptions, intOptions, boolOptions);
 
         static constexpr const char *AGENT_OBJECT_PATH = "/com/pelagicore/SoftwareContainerAgent";
-        DBus::Connection connection = getBusConnection(useSessionBus);
-        SoftwareContainerAgent agent(mainContext, config);
-        std::unique_ptr<SoftwareContainerAgentAdaptor> adaptor =
-            std::unique_ptr<SoftwareContainerAgentAdaptor>(
-                new DBusCppAdaptor(connection, AGENT_OBJECT_PATH, agent)
-            );
+        ::softwarecontainer::SoftwareContainerAgent agent(mainContext, config);
```
Comment (tobsan) [inline agent/src/main.cpp:206]: Do you really need the prefix :: here?

File: agent/src/softwarecontaineragent_dbus_stub.cpp [hunk 2]
```diff
@@ -0,0 +1,322 @@
+#include "softwarecontaineragent_dbus_stub.h"
+
+Glib::ustring interfaceXml0 = R"xml(
```
Comment (tobsan) [inline agent/src/softwarecontaineragent_dbus_stub.cpp]: Does this really work? Using "xml" as delimiter should stop processing on the first line I think. Even if it happens to work, choose something that does not show up in the xml code, like "endOfXMLCode".

**Review by tobsan — CHANGES_REQUESTED** [review]
Looks really good overall, see inline comments.

File: agent/src/main.cpp (hunk 1 above)
Comment (DunderRoffe) [reply agent/src/main.cpp:206]: yes, we use namespaces in a super wierd way that forces this.

File: agent/src/softwarecontaineragentadaptor.cpp [hunk 3]
```diff
@@ -6,80 +6,96 @@ SoftwareContainerAgentAdaptor::~SoftwareContainerAgentAdaptor()
...
 
-SoftwareContainerAgentAdaptor::SoftwareContainerAgentAdaptor(SoftwareContainerAgent &agent) :
-    m_agent(agent)
+SoftwareContainerAgentAdaptor::SoftwareContainerAgentAdaptor(::softwarecontainer::SoftwareContainerAgent &agent, bool useSessionBus) :
+    com::pelagicore::SoftwareContainerAgent(), m_agent(agent)
 {
+    std::string agentBusName = "com.pelagicore.SoftwareContainerAgent";
+    Gio::DBus::BusType busType = useSessionBus ? Gio::DBus::BUS_TYPE_SESSION
+                                               : Gio::DBus::BUS_TYPE_SYSTEM;
+    connect(busType, agentBusName);
```
Comment (tobsan) [inline agent/src/softwarecontaineragentadaptor.cpp:15]: are there any risks when running connect here in the constructor?

**Review by tobsan — APPROVED** [review]

File: agent/src/softwarecontaineragentadaptor.cpp (hunk 3 above)
Comment (DunderRoffe) [reply agent/src/softwarecontaineragentadaptor.cpp:15]: Not really. `connect` is supposed to handles possible exceptions internally.

**Pull request merged**